packages/next/src/build/swc/options.ts

```diff
@@ -9,7 +9,7 @@ import type { ResolvedBaseUrl } from '../load-jsconfig'
 import { isWebpackServerOnlyLayer, isWebpackAppPagesLayer } from '../utils'
 
 const nextDistPath =
-  /(next[\\/]dist[\\/]shared[\\/]lib)|(next[\\/]dist[\\/]client)|(next[\\/]dist[\\/]pages)/
+  /([\\/]next[\\/]dist[\\/]shared[\\/]lib)|([\\/]next[\\/]dist[\\/]client)|([\\/]next[\\/]dist[\\/]pages)/
```

Comment:
maybe even check `([\\/]node_modules|[\\/]next.js[\\/]packages)[\\/]next`

```suggestion
  /([\\/]node_modules|[\\/]next.js[\\/]packages)[\\/]next[\\/]dist[\\/](shared[\\/]lib|client|pages)/
```

Reply:
I've used `path.dirname(require.resolve('next/package.json'))` to look up the next dir.